### DISCUSSION TITLE
Why aren't books free?

### DISCUSSION THREAD

**Discussion by kuanbase** [vlang/v#22238] [General]
Because Rust is a bunch of free tutorials.

**Comment by Wajinn** [comment]
Possibly the below could be of some help, though not exactly clear if you are only looking for books or any tutorial to help learn.

First, to clear-up some things:

1) Books about Rust, also cost money. This is easy for us to see, by looking on Amazon. They are not all free.
2) V is a still developing language, where Rust is many years older, and was created within a large American company.
3) Many other young languages don't have any books about them, where V already has 4 published books about it.
4) V has documentation about the [language](https://docs.vlang.io/) and its [modules](https://modules.vlang.io/).
5) There is a [gitbook](https://lydiandylin.gitbook.io/vlang) (by V contributor- [lydiandy](https://github.com/lydiandy)), based on V's documentation, that's translated to Chinese (appears to be your native language).

There are also some YouTube videos that give an overview of V or are tutorials (and have subtitles in different languages):

Note: All have Chinese subtitles (and other languages) via YouTube's auto-translate
1) [A small presentation of V's features at IBM](https://youtu.be/_dIAcNp9bJs?si=AE6zwEnCQHu1ycCy)
2) [An introduction to V - the vlang](https://youtu.be/ZO3vlwqvTrc?si=Ssl0Lc7xOZQcyqU6) 
3) [Read Fast, Write Fast, Run Faster: How To Maintain And Iterate With V](https://youtu.be/pchEsYwA74Q?si=kwbHBgZGqXQuOOLD)
4) [Aprendendo a Linguagem V/Learning the V Language](https://www.youtube.com/playlist?list=PLpz113t3_XTuEF9jpDxvvWatK9BLBbb6W) 
5) [V programming by coderlyfe](https://www.youtube.com/playlist?list=PLvLz9K34Y26baplmBi-3oJNA7gzb5Nqui) (little old, but good)

Links to code examples in V:

1) [V on Exercism](https://exercism.org/tracks/vlang)
2) [V on Rosetta Code](https://rosettacode.org/wiki/Category:V_(Vlang))
3) [V examples](https://github.com/vlang/v/tree/master/examples)

Possibly in the not so distant future, there might be an English gitbook or wikibook, based on documentation. And hopefully more published books about V on Amazon, based on different subjects or to be used as an educational/CS reference.
